streams/src/main/java/com/palantir/common/streams/MoreCollectors.java

```diff
@@ -61,6 +61,18 @@ public final class MoreCollectors {
                 ImmutableList.Builder::build);
     }
 
+    /**
+     * Collect a Stream of Map.Entry (e.g. a StreamEx EntryStream) into a Guava {@link ImmutableMap}, which preserves
+     * iteration order. Duplicate keys will result in an error. Throws NullPointerException if any key or value is null.
+     *
+     * For behaviour details, see docs on {@link ImmutableMap#toImmutableMap}.
+     *
+     * Beware that {@code EntryStream#toImmutableMap()} does NOT preserve iteration order, as it uses a regular HashMap.
+     */
+    public static <K, V> Collector<Map.Entry<K, V>, ?, ImmutableMap<K, V>> toImmutableMap() {
```

Comment:
One potential API edge case:

`toImmutableMap` is generic, and based on a stream of `Map.Entry`. It's possible we may want to add over `toImmutableMap` overloads from other types in the future. Is that realistic/likely? Probably not, I'm not really sure.
We could name this `entriesToImmutableMap` or similar to avoid that shape of problem in the future, or leave it as is.

I don't have a strong opinion here, just thinking through it